syntax = "proto3";

import "graph.proto";

package serialization_data;

message Settings {
  int32 bus_wait_time = 1;
  double bus_velocity = 2;
}

message RouteInternalData {
  double weight = 1;
  int32 prev_edge = 2;
  bool has_prev_edge_ = 3;
  bool has_internal_data = 4;
};

message InternalData {
  repeated RouteInternalData internal_data = 1;
}

message Activity {
  int32 number = 1;
  int32 from = 2;
  uint32 span_count = 3;
  double time = 4;
  uint32 vertexId = 5;
}

message Router {
  Settings router_settings = 1;
  Graph graph = 2;
  repeated InternalData data = 3;
  repeated Activity activity = 4;
}
